openedx/core/djangoapps/course_groups/migrations/0006_cohort_membership_data_migrate.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+
+class Migration(DataMigration):
+
+    def forwards(self, orm):
+        # Matches CourseUserGroup.COHORT
+        cohort_type = 'cohort'
+
+        for cohort_group in orm.CourseUserGroup.objects.all():
+            for user in cohort_group.users.all():
+                current_course_groups = orm.CourseUserGroup.objects.filter(
+                    course_id=cohort_group.course_id,
+                    users__id=user.id,
+                    group_type=cohort_type
+                )
+                current_user_groups = user.course_groups.filter(
+                    course_id=cohort_group.course_id,
+                    group_type=cohort_type
+                )
+
+                unioned_set = set(current_course_groups).union(set(current_user_groups))
+
+                # Per product guidance, fix problem users by arbitrarily choosing a single membership to retain
+                cohort_itr = unioned_set.pop()
```

Comment:
I'm glad you spotted this - shouldn't we be saving this value? When we create the actual membership a few lines below, we should be using this value instead of the outer loop iterator `cohort_group`.